### PR TITLE
Get tenancy action diary details usecase

### PR DESCRIPTION
**Pull request by LBHRShetty**
This PR includes:
Added WhenGivenATenancyRef_Index_ShouldRespondWithTenancyActionDiaryInfoForThatTenancy in the controller test.
Added stub, implement new function signature **ExecuteActionDiaryQuery** in interface  IListTenancies
The test now passes the assertion test.

### CODE REVIEW

File: LBHTenancyAPI/Controllers/TenanciesController.cs [hunk 1]
```diff
@@ -48,5 +48,34 @@ public TenanciesController(IListTenancies listTenancies)
 
             return Ok(result);
         }
+
+
```
Comment (VladyslavAtamanyuk) [inline LBHTenancyAPI/Controllers/TenanciesController.cs]: Wild White space

File: LBHTenancyAPI/Gateways/ITenanciesGateway.cs [hunk 2]
```diff
@@ -6,5 +6,6 @@ namespace LBHTenancyAPI.Gateways
     public interface ITenanciesGateway
     {
         List<TenancyListItem> GetTenanciesByRefs(List<string> tenancyRefs);
+        List<ArrearsActionDiaryDetails> GetActionDiaryDetailsbyTenancyRef(List<string> actionDiary);
```
Comment (VladyslavAtamanyuk) [inline LBHTenancyAPI/Gateways/ITenanciesGateway.cs]: should we rename "actionDiary" to TenancyRef as we are getting action diary details by tenancy reference?

Comment (LBHRShetty) [reply LBHTenancyAPI/Gateways/ITenanciesGateway.cs]: Good point!

File: LBHTenancyAPI/Gateways/StubTenanciesGateway.cs [hunk 3]
```diff
@@ -19,10 +21,15 @@ public List<TenancyListItem> GetTenanciesByRefs(List<string> tenancyRefs)
             {
                 tenancies.Add(StoredTenancyListItems[tenancyRef]);
             }
-
             return tenancies;
         }
 
+        public List<ArrearsActionDiaryDetails> GetActionDiaryDetailsbyTenancyRef(List<string> actionDiaryDetail)
+        {
```
Comment (VladyslavAtamanyuk) [inline LBHTenancyAPI/Gateways/StubTenanciesGateway.cs]: rename the actionDiaryDetail string here to TenacyRef as before.

File: LBHTenancyAPI/Gateways/UhTenanciesGateway.cs [hunk 4]
```diff
@@ -128,29 +127,20 @@ public List<ArrearsActionDiaryDetails> GetLatestfiveArrearsActionForRef(string t
 
         }
 
-            public Tenancy GetLatestfiveArrearsActionForRef(string tenancyRef)
+        public List<ArrearsActionDiaryDetails> GetActionDiaryDetailsbyTenancyRef(List<string> tenancyRefs)
```
Comment (VladyslavAtamanyuk) [inline LBHTenancyAPI/Gateways/UhTenanciesGateway.cs]: Not sure if we are using one reference here or many but the method imply we are using one but we are passing a list of many references, we need to rename one of these so it is clear.

File: LBHTenancyAPITest/Test/Controllers/TenanciesTest.cs [hunk 5]
```diff
@@ -146,13 +150,71 @@ public class TenanciesTest
             Assert.Equal(expectedJson, actualJson);
         }
 
+
```
Comment (VladyslavAtamanyuk) [inline LBHTenancyAPITest/Test/Controllers/TenanciesTest.cs]: Wild White space

File: LBHTenancyAPI/Gateways/UhTenanciesGateway.cs (hunk 4 above)
Comment (LBHRShetty) [reply LBHTenancyAPI/Gateways/UhTenanciesGateway.cs]: Yes sorry I missed 's' in the name.It takes list of tenancyRef

**Pull request merged**